# This patchfile modifies src/Makefile to allow for extra libraries to be
# added to lib[cdsz]mumps.so (e.g,. libmumps_common.so and possible libmpiseq.so)
#
# It also adds flags to produce PIC code

diff -Naur spack-src/src/Makefile spack-src.patched/src/Makefile
--- spack-src/src/Makefile	2019-04-18 05:55:07.000000000 -0400
+++ spack-src.patched/src/Makefile	2020-10-31 15:19:49.927297524 -0400
@@ -182,7 +182,7 @@
 	$(RANLIB) $@
 
 $(libdir)/lib$(ARITH)mumps$(PLAT)$(LIBEXT):    $(OBJS_MOD) $(OBJS_OTHER)
-	$(AR)$@ $?
+	$(AR)$@ $? $(EXTRA_LIBS4MUMPS)
 	$(RANLIB) $@
 
 # Dependencies between modules:
@@ -378,9 +378,9 @@
 
 .SUFFIXES: .c .F .o
 .F.o:
-	$(FC) $(OPTF) $(INCS) $(IORDERINGSF) $(ORDERINGSF) -I. -I../include -c $*.F $(OUTF)$*.o
+	$(FC) $(OPTF) $(FC_PIC_FLAG) $(INCS) $(IORDERINGSF) $(ORDERINGSF) -I. -I../include -c $*.F $(OUTF)$*.o
 .c.o:
-	$(CC) $(OPTC) $(INCS) -I../include $(CDEFS) $(IORDERINGSC) $(ORDERINGSC) -c $*.c $(OUTC)$*.o
+	$(CC) $(OPTC) $(CC_PIC_FLAG) $(INCS) -I../include $(CDEFS) $(IORDERINGSC) $(ORDERINGSC) -c $*.c $(OUTC)$*.o
 
 $(ARITH)mumps_c.o:	mumps_c.c
 	$(CC) $(OPTC) $(INCS) $(CDEFS) -DMUMPS_ARITH=MUMPS_ARITH_$(ARITH) \
